crates/surrealdb-types/src/value/array.rs

```diff
@@ -0,0 +1,112 @@
+use std::ops::{Deref, DerefMut};
+
+use serde::{Deserialize, Serialize};
+
+use crate::{SurrealValue, Value};
+
+/// Represents an array of values in SurrealDB
+///
+/// An array is an ordered collection of values that can contain elements of any type.
+/// The underlying storage is a `Vec<Value>`.
+#[derive(Clone, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize, Deserialize)]
+pub struct Array(pub(crate) Vec<Value>);
+
+impl Array {
+	/// Create a new empty array
+	pub fn new() -> Self {
+		Array(Vec::new())
+	}
+	/// Create a new array with capacity
+	pub fn with_capacity(len: usize) -> Self {
+		Self(Vec::with_capacity(len))
+	}
+	/// Get the length of the array
+	pub fn len(&self) -> usize {
+		self.0.len()
+	}
+	/// Check if there array is empty
+	pub fn is_empty(&self) -> bool {
+		self.0.is_empty()
+	}
+}
+
+impl<T: SurrealValue> From<Vec<T>> for Array {
+	fn from(v: Vec<T>) -> Self {
+		v.into_iter().map(T::into_value).collect()
+	}
+}
+
+impl From<Array> for Vec<Value> {
+	fn from(s: Array) -> Self {
+		s.0
+	}
+}
+
+impl<T: SurrealValue> FromIterator<T> for Array {
+	fn from_iter<I: IntoIterator<Item = T>>(iter: I) -> Self {
+		Array(iter.into_iter().map(T::into_value).collect())
+	}
+}
+
+impl Deref for Array {
+	type Target = Vec<Value>;
+	fn deref(&self) -> &Self::Target {
+		&self.0
+	}
+}
+
+impl DerefMut for Array {
+	fn deref_mut(&mut self) -> &mut Self::Target {
+		&mut self.0
+	}
+}
+
+impl IntoIterator for Array {
+	type Item = Value;
+	type IntoIter = std::vec::IntoIter<Self::Item>;
+	fn into_iter(self) -> Self::IntoIter {
+		self.0.into_iter()
+	}
+}
+
+// ------------------------------
+
+impl std::ops::Add<Value> for Array {
+	type Output = Self;
+	fn add(mut self, other: Value) -> Self {
+		self.0.push(other);
+		self
+	}
+}
+
+impl std::ops::Add for Array {
+	type Output = Self;
+	fn add(mut self, mut other: Self) -> Self {
+		self.0.append(&mut other.0);
+		self
+	}
+}
+
+// ------------------------------
+
+impl std::ops::Sub<Value> for Array {
+	type Output = Self;
+	fn sub(mut self, other: Value) -> Self {
+		if let Some(p) = self.0.iter().position(|x| *x == other) {
+			self.0.remove(p);
+		}
+		self
+	}
+}
+
+impl std::ops::Sub for Array {
+	type Output = Self;
+	fn sub(mut self, other: Self) -> Self {
+		for v in other.0 {
+			if let Some(p) = self.0.iter().position(|x| *x == v) {
+				self.0.remove(p);
+			}
+		}
+		self
+	}
+}
```

Comment:
I don't think these traits should be implemented. You should be restrictive when implementing operator overloading. It is not immediately obvious what `Array + Value` will do, it could be multiple things from pushing the value at the beginning, end or adding the value to every element of the array. Therefore these traits should not be implemented.